js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx

```diff
@@ -0,0 +1,335 @@
+        <Box flex={{direction: 'row', justifyContent: 'center'}} style={{paddingTop: '100px'}}>
+          <Box flex={{direction: 'row', alignItems: 'center', gap: 16}}>
+            <Spinner purpose="body-text" />
+            <div style={{color: Colors.textLight()}}>Loading assets…</div>
+          </Box>
+        </Box>
+      </Page>
+    );
+  }
+
+  return (
+    <Box flex={{direction: 'column'}} style={{height: '100%', overflow: 'hidden'}}>
+      <AssetPageHeader
+        assetKey={{path: currentPath}}
+        right={
+          <Box flex={{gap: 12, alignItems: 'center'}}>
+            <ReloadAllButton label="Reload definitions" />
+          </Box>
+        }
+      />
+      <Box
+        padding={64}
+        flex={{justifyContent: 'center', alignItems: 'center'}}
+        style={{overflowY: 'scroll'}}
+      >
+        <Box style={{width: '60%'}} flex={{direction: 'column', gap: 16}}>
+          <Box flex={{direction: 'row', alignItems: 'center', justifyContent: 'space-between'}}>
+            <Heading>
+              {getGreeting()}
+              {viewerName ? `, ${viewerName}` : ''}
+            </Heading>
+            <Box flex={{direction: 'row', gap: 12, alignItems: 'center'}}>
+              <ViewAllAssetsLink />
+              <AssetGlobalLineageButton />
+            </Box>
+          </Box>
+          <TextInput />
+        </Box>
+      </Box>
+      <Box flex={{direction: 'column'}}>
+        {Object.keys(assetCountBySection.countsByOwner).length > 0 && (
+          <>
+            <SectionHeader sectionName="Owner" />
+            <SectionBody>
+              {Object.entries(assetCountBySection.countsByOwner).map(([label, count], idx) => (
+                <CountForAssetType key={idx} assetsCount={count}>
+                  <UserDisplay key={idx} email={label} size="normal" />
+                </CountForAssetType>
+              ))}
+            </SectionBody>
+          </>
+        )}
+        {Object.keys(assetCountBySection.countsByComputeKind).length > 0 && (
+          <>
+            <SectionHeader sectionName="Compute kind" />
+            <SectionBody>
+              {Object.entries(assetCountBySection.countsByComputeKind).map(
+                ([label, count], idx) => (
+                  <CountForAssetType key={idx} assetsCount={count}>
+                    <Box flex={{direction: 'row', gap: 4, alignItems: 'center'}}>
+                      <TagIcon tag_label={label} />
+                      <Link to={LinkToAssetGraphComputeKind(label)}>{label}</Link>
+                    </Box>
+                  </CountForAssetType>
+                ),
+              )}
+            </SectionBody>
+          </>
+        )}
+        {assetCountBySection.countPerAssetGroup.length > 0 && (
+          <>
+            <SectionHeader sectionName="Asset groups" />
+            <SectionBody>
+              {assetCountBySection.countPerAssetGroup.map(
+                (assetGroupCount: CountPerGroupName, idx) => (
```

Comment:
You probably shouldn't have to annotate this type, TS should be able to infer it.